src/number_theory/chicken_mcnugget.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2021 Alex Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Zhao
+-/
+import data.nat.modeq
+import group_theory.submonoid.basic
+import group_theory.submonoid.membership
+
+/-!
+# Chicken McNugget Theorem
+
+In this file we prove the Chicken McNugget Theorem, first using natural multiplication,
+then using additive submonoid closures. This theorem solves the 2-variable variant of the
+general Frobenius coin problem
+
+## Theorem Statement
+The Chicken McNugget Theorem states, for two relatively prime integers `m` and `n`, both larger
+than 1, the greatest integer not expressible as a sum of nonnegative multiples of these two
+is `m * n - m - n`. The general problem of finding this greatest integer for any number of
+(not pairwise) relatively prime integers is called the Frobenius coin problem.
+
+## Implementation Notes
+
+For the upper bound, we begin with an auxiliary lemma showing `m * n` is not attainable, then show
+`m * n - m - n` is not attainable. Then for the construction, we create a `k_1` which is `k mod n`
+and `0 mod m`, then show it is at most `k`. Then `k_1` is a multiple of `m`, so `(k-k_1)`
+is a multiple of n, and we're done.
+
+Afterwards, we rewrite this with add_submonoid.closure and is_greatest,
+using `add_submonoid.mem_closure_pair`.
+
+## Tags
+
+chicken mcnugget, frobenius coin, chinese remainder theorem, submonoid.closure
+-/
+
+namespace nat
+
+/- No solution for the product over the natural numbers. -/
+lemma mul_add_mul_ne_mul_of_coprime {a b m n : ℕ} (ha : a ≠ 0) (hb : b ≠ 0)
```

Comment:
You could add `variables {m n : ℕ}` or `variables {a b m n : ℕ}` at the top of the file.